java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/MultiSizeSlidingWindowKeyedProcessFunction.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2022 The Feathub Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.feathub.flink.udf;
+
+import org.apache.flink.api.common.functions.RuntimeContext;
+import org.apache.flink.api.common.state.MapState;
+import org.apache.flink.api.common.state.MapStateDescriptor;
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+import org.apache.flink.api.common.typeutils.base.LongSerializer;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.KeyedProcessFunction;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.CollectionUtil;
+import org.apache.flink.util.Collector;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Map;
+
+/**
+ * A KeyedProcessFunction that aggregate sliding windows with different sizes. The ProcessFunction
+ * should be applied after the input has been aggregated with a tumble window with size equals to
+ * the step size of sliding window. With this process function, we only need to keep the rows for
+ * the maximum window size, instead of rows for each sliding window.
+ *
+ * <p>The ProcessFunction assumes that: 1. rows of each key are ordered by the row time 2. row time
+ * attribute of rows with the same key are all distinct. The assumptions hold true after applying
+ * the tumbling window aggregation whose window size is the same as the step size of the {@link
+ * MultiSizeSlidingWindowKeyedProcessFunction} to be applied.
+ */
+public class MultiSizeSlidingWindowKeyedProcessFunction
```

Comment:
Can we make the class package private (i.e. remove the `public` keyword) since it is supposed to be used only by MultiSizeSlidingWindowUtils?

Can we reorganize the files so that the public classes (e.g. MultiSizeSlidingWindowUtils, PostSlidingWindowUtils) are at higher level than package private classes?